pythran/optimizations/iter_transformation.py

```diff
@@ -59,15 +59,21 @@ def find_matching_builtin(self, node):
                 return keyword
 
     def visit_Module(self, node):
-        """ Add itertools import for imap, izip or ifilter iterator. """
+        """Add itertools import for imap, izip or ifilter iterator."""
         self.generic_visit(node)
         importIt = ast.Import(names=[ast.alias(name='itertools', asname=None)])
         return ast.Module(body=([importIt] + node.body))
 
     def visit_Call(self, node):
-        """ Replace function call by its correct iterator if it is possible."""
+        """Replace function call by its correct iterator if it is possible."""
         if node in self.potential_iterator:
             match_keyword = self.find_matching_builtin(node)
+            # Special handling for map which can't be turn to imap with None as
+            # a parameter as map(None, [1, 2]) == [1, 2] while
+            # list(imap(None, [1, 2])) == [(1,), (2,)]
+            if match_keyword == "map":
+                if MODULES["__builtin__"]["None"] in self.aliases[node.args[0]]:
+                    return self.generic_visit(node)
```

Comment:
I think it's safer to check that aliases are functiondef. One could alias to a parameter that could be none, and in that case better be conservative.